src/Sql/Sql/ServerTrustGroup/Cmdlet/NewAzureSqlServerTrustGroup.cs

```diff
@@ -0,0 +1,120 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.ServerTrustGroup.Model;
+using Microsoft.Rest.Azure;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Text;
+
+namespace Microsoft.Azure.Commands.Sql.ServerTrustGroup.Cmdlet
+{
+	/// <summary>
+	/// Cmdlet to create a new Azure Sql Server Trust Group.
+	/// </summary>
+	[Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlServerTrustGroup", DefaultParameterSetName = "Default"), OutputType(typeof(AzureSqlServerTrustGroupModel))]
+	public class NewAzureSqlServerTrustGroup : AzureSqlServerTrustGroupCmdletBase
+	{
+		/// <summary>
+		/// Gets or sets the name of the resource group to use.
+		/// </summary>
+		[Parameter(Mandatory = true,
+			Position = 0,
+			HelpMessage = "The name of the resource group.")]
+		[ResourceGroupCompleter]
+		[ValidateNotNullOrEmpty]
+		public override string ResourceGroupName { get; set; }
+
+		/// <summary>
+		/// Gets or sets the location of the ServerTrustGroup to use.
+		/// </summary>
+		[Parameter(Mandatory = true,
+			Position = 1,
+			HelpMessage = "The location of the Server Trust Group to create.")]
+		[LocationCompleter("Microsoft.Sql/locations/serverTrustGroups")]
+		[ValidateNotNullOrEmpty]
+		public string Location { get; set; }
+
+		/// <summary>
+		/// Gets or sets the name of the ServerTrustGroup to use.
+		/// </summary>
+		[Parameter(Mandatory = true,
+			Position = 2,
+			HelpMessage = "The name of the Server Trust Group to create.")]
+		[ValidateNotNullOrEmpty]
+		public string Name { get; set; }
+
+		/// <summary>
+		/// Gets or sets group members of the ServerTrustGroup to use.
+		/// </summary>
+		[Parameter(Mandatory = true,
+			Position = 3,
+			HelpMessage = "Group members of the Server Trust Group to create.")]
+		[ValidateNotNullOrEmpty]
+		public List<String> GroupMembers { get; set; }
+
+		/// <summary>
+		/// Gets or sets the trust scope of the ServerTrustGroup to use.
+		/// </summary>
+		[Parameter(Mandatory = false,
+			Position = 4,
+			HelpMessage = "The trust scope of the Server Trust Group to create.")]
+		[ValidateNotNullOrEmpty]
+		public List<String> TrustScope { get; set; }
+
+		protected override IEnumerable<AzureSqlServerTrustGroupModel> GetEntity()
+		{
+			try
+			{
+				ModelAdapter.GetServerTrustGroup(this.ResourceGroupName, this.Location, this.Name);
+			}
+			catch (CloudException ex)
+			{
+				if (ex.Response.StatusCode == System.Net.HttpStatusCode.NotFound)
+				{
+					return null;
+				}
+
+				throw;
+			}
+
+			throw new PSArgumentException("This Server Trust Group already exists");
```

Comment:
Consider adding more info to the exception message, such as Server trust group name.